kubeflow/trainer/utils/utils.py

```diff
@@ -318,12 +320,44 @@ def get_command_using_train_func(
     # Wrap function code to execute it from the file. For example:
     # TODO (andreyvelich): Find a better way to run users' scripts.
     # def train(parameters):
-    #     print('Start Training...')
+    #   print('Start Training...')
     # train({'lr': 0.01})
     if train_func_parameters is None:
-        func_code = f"{func_code}\n{train_func.__name__}()\n"
+        func_call = f"{train_func.__name__}()"
     else:
-        func_code = f"{func_code}\n{train_func.__name__}({train_func_parameters})\n"
+        # Decide whether to pass a single dict or unpack as kwargs using the same hinting logic.
+        sig = inspect.signature(train_func)
+        params = list(sig.parameters.values())
+        try:
+            from typing import get_type_hints
+            type_hints = get_type_hints(train_func)
+        except Exception:
+            type_hints = {}
+
+        call_with_single_dict = (
+            len(params) == 1
+            and (
+                type_hints.get(params[0].name, dict) is dict
+                or params[0].annotation is inspect.Parameter.empty
+            )
+        )
+
+        if call_with_single_dict:
+            message = (
+                f"Calling '{train_func.__name__}' with a single dict argument is deprecated."
+                " Please update your function to accept individual keyword arguments."
+            )
+            warnings.warn(
+                message,
+                DeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
We can put the msg directly to the warning.

```suggestion
            warnings.warn(
                f"Calling '{train_func.__name__}' with a single dict argument is deprecated."
                " Please update your function to accept individual keyword arguments."
                DeprecationWarning,
                stacklevel=2,
            )
```